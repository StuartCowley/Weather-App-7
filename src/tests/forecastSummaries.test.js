import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../components /forecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecast: [
      {
        date: 1111,
        description: " Stub description 1.0",
        icon: 800,
        temperature: {
          min: 12,
          max: 22,
        },
      },
      {
        date: 3333,
        description: " Stub description 2.0",
        icon: 602,
        temperature: {
          min: 1,
          max: 10,
        },
      },
    ],
    onForecastSelect: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecast={validProps.forecast}
        onForecastSelect={validProps.onForecastSelect}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries
        forecast={validProps.forecast}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
