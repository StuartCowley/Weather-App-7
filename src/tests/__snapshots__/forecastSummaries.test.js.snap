// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu 4th Jan
      </div>
      <div
        class="forecast-summary_description"
      >
         Stub description 1.0
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        22°C
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu 4th Jan
      </div>
      <div
        class="forecast-summary_description"
      >
         Stub description 2.0
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-snow"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        10°C
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
